src/prototype_creep_routing.js

```diff
@@ -219,15 +228,17 @@ Creep.prototype.moveByPathMy = function(path, pathPos, directions) {
   this.buildRoad();
 
   if (!validateDirections(directions)) {
-    this.log(`moveByPathMy: Directions invalid pathPos: ${pathPos} path: ${path} directions: ${directions}`);
-    return false;
+    if (config.debug.routing) {
+      this.log(`${Game.time} moveByPathMy: Directions invalid pathPos: ${pathPos} path[pathPos]: ${path[pathPos]} directions: ${global.ex(directions, 1)}`);
+    }
+    return true;
```

Comment:
because you return in followPath (prototype_creep_routing.js:175) moveByPathMy, and that leads to unnessesary 'Directions invalid messages' for carries...

Reply:
because of the skipped rooms they are on the wrong room side

Reply:
so it lead to 'Reached end of handling() why?' for carries and some scourcers